p2p/discovery/mdns.go

```diff
@@ -69,12 +88,11 @@ func getDialableListenAddrs(ph host.Host) ([]*net.TCPAddr, error) {
 	return out, nil
 }
 
-func NewMdnsService(ctx context.Context, peerhost host.Host, interval time.Duration, serviceTag string) (Service, error) {
-
+func NewMdnsService(ctx context.Context, peerhost host.Host, interval time.Duration, serviceTag string, iface *net.Interface) (Service, error) {
```

Comment:
This is a breaking change. Seems like we could avoid this using options varargs (https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis). WDYT @aarshkshah1992

Reply:
Ideally if we went the option varargs route the interval would not be mandatory so if we're going to be doing a breaking change we should probably do that one.

Reply:
@aschmahmann Agreed on both points.